Hardware/Hardware/CardReader/CardReader.swift

```diff
@@ -1,7 +1,18 @@
 /// Models a Card Reader. This is the public struct that clients of
 /// Hardware are expected to consume.
-/// The exact properties are to be defined yet.
-/// For now, this is a placeholder
 public struct CardReader {
+    public let serial: String
+    public let vendorIdentifier: String?
+    public let name: String?
     public let status: CardReaderStatus
+    public let softwareVersion: String?
+    public let batteryLevel: Float?
+    public let readerType: CardReaderType
+}
+
+
+extension CardReader: Equatable {
+    public static func ==(lhs: CardReader, rhs: CardReader) -> Bool {
+        lhs.serial == rhs.serial
+    }
```

Comment:
Hmm... I think implementing equatable on value-types by only checking one property could lead to some errors in the future.  

What do you think about just adding a method, `hasSameSerial()` or similar.

Reply:
> I think implementing equatable on value-types by only checking one property could lead to some errors in the future.

I'm curious, why? What difference would it make to add that layer of indirection?

Reply:
To me, the expectation when I do a `==` over two value-types is that that object will behave the same as the other because they are "equals". Here, by comparing only one property, we can't have that expectation about it. Operations would likely have different outcomes because the object can have different values.

Additionally, I can see ourselves getting some false positives on unit tests because someone used `XCTAssertEqual` expecting the objects to be exactly the same.